Edit screen

CatalinDavid17 opened the pull request.

Commit:
Added CDEditTableViewController in controller and its class
Added prototypeCells with their constraints(might suffer additional changes)

Commit:
Merge branch 'custom_cell_and_core_data' into TemporaryEditScreen

* custom_cell_and_core_data:
  - Modified date, note, and priority labels name into dateLabel, noteLabel, priorityLabel - Added NS_ENUM (CDPriorityLow, CDPriorityMedium, CDPriorityHigh, CDPriorityCritical) - Added an external enum which gives the string according to the enum element - modified changeProperties - changePropertiesForReminder - modified [NSNumber numberWithInteger] with proper use of NS_ENUM elements

Commit:
Added classes for the custom cells

Commit:
Merge branch 'TemporaryEditScreen' into EditScreen

* TemporaryEditScreen:
  Added classes for the custom cells
  Added CDEditTableViewController in controller and its class Added prototypeCells with their constraints(might suffer additional changes)

Commit:
Added outlets: - CDEditReminderTextCell -> textView
			 - CDEditPriorityCell -> prioritySegmentedControl
			 - CDEditNotesTextCell -> noteTextView
			 - CDEditDatePickedCell -> datePicker

Commit:
Edit Screen Updated - Dynamic cells - with functionality on switch button for inserting Alert and Repeat cell

Commit:
Temporary

Commit:
- Added CDEditReminderTextCell - injected selected reminder, edit and save on unwind

Commit:
Partial Implementation : - Added reminderText functionality - assignate the value of the textView by refference into an injected reminder
					- Attempt to save it into reminderController in Unwind method
					- Added priority + notes functionality - same as above.
					- removed gray color for selected cells - reminderTextCell + noteCell

Commit:
- renamed some properties for a better understanding

Commit:
- removed selected gray for reminderOnDayCell

Commit:
- removed priorityCell selected Gray

Commit:
- priorityCell removed selectedGray

Commit:
- Added DatePickerCell to insert and delete on select cell - with animation, change color and date format

Commit:
- Added functionality to DatePickerCell - saving date to the tapped reminder

Commit:
- fixed code dateFormatter

Commit:
- added reloadOnRow for updating reminder after comming from editing screen

Commit:
- code review

Commit:
- fixed reminder cell updating when comming back from editing. In the case when we have note plus date, there was a constant set to a minimum value,
which was never increased when we have both.

Commit:
- rewritten code, using literals for NSNumber type in the priority enum

Commit:
- Added functionality to detailButtons - for both cells changed and working properly
- Reworked code due to analyzer warnings
- Added code from edit screen in cellForRowAtIndexPath to each custom cell class

Commit:
- added placeholder for note in edit screen

Misantrofia commented on ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDReminderTableViewController.m:
```diff
@@ -11,12 +11,14 @@
 #import "CDReminderCell.h"
```
Observed during testing. The add reminder cell shouldn't have the option to swipe and delete. 
- If you add a lot of reminders the tableview does not scroll and the add cell becomes off screen. Add functionality to scroll to last when a new reminder is added

Misantrofia commented on ReminderApp-OBJC/ReminderApp-OBJC/Classes/ReminderCell/CDEditPriorityCell.h:
```diff
@@ -0,0 +1,25 @@
+#import <UIKit/UIKit.h>
+#import "CDReminder.h"
+
+typedef NS_ENUM(NSInteger, CDPriority){
```
This should go in the Reminder class, it is related to the model. If you need this enum in some other class you will have to import CDEditPriorityCell.h which is not ok.

Misantrofia commented on ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDReminderTableViewController.m:
```diff
@@ -136,12 +152,24 @@ - (void)reminderCell:(CDReminderCell *)reminderCell wantsToResizeTextView:(UITex
+}
+
+- (void)addReminderCell:(CDAddReminderCell *)addRemindercell wantsToAddReminderWithText:(NSString *)reminderText
+		viaDetailButton:(BOOL)detailButton{
```
Change viaDetailButton to "detailButtonWasPressed"

Commit:
- moved CDPriority enum to the reminder model class
- fixed remove placeholder when begin editing on the reminder title from the edit screen

Commit:
Changed viaDetailButton with detailButtonWasPressed - *addReminderCell delegate method
- Changed the logic for addReminderCell placeholder

Commit:
- placeholder addReminderCell - fixed one case

Commit:
- comment

Commit:
- added scroll to the last cell (addReminderCell)

Commit:
- partial fix for footer

Commit:
- removed addReminderCell
- partial implementation on footer cell - working functionality(not resizable yet)
- fixed flicker footer, fixed flicker when passing on newRow from reminderCell

Commit:
space

Commit:
renamed

Commit:
- added shadow handling - on/off - if the tableView is scrolled to bottom or not

Commit:
- Added to EditScreen - listCell + its functionality
- Added CDChangeTopicForReminder
- Added delegate: CDTopicTableViewControllerDelegate - changeTopicController will conform to this delegate
											     - added readyToSendTopicList - once changeTopicController is loaded, we set topic's delegate and perform topicListInjection
- added changedTopic block, which will change the selected reminder's topic

Commit:
- fixed - update to changed list title when return from changeListController
 - fixed - shadow when initially a screen is loaded and its visible cells is lower than table's offset

Commit:
changes

Commit:
- added functionality for detail button - appear and dissapear when there is/not content in the addReminderCell

Commit:
* EditScreen:
  - added functionality for detail button - appear and dissapear when there is/not content in the addReminderCell
  - fixed - update to changed list title when return from changeListController  - fixed - shadow when initially a screen is loaded and its visible cells is lower than table's offset

Commit:
- added Local notification settings method

Commit:
- added notification observer
- added notification handler
- added snooze option

Commit:
- added textColor to priorityLabel + notesLabel

Commit:
- changed alpha to dateLabel.textColor

Commit:
- made snooze notification require no authentication

Commit:
- renamed notification identifiers

Commit:
- added logs when notifications has been handled
- removed - changes to reminder date, when hitting snooze

Commit:
- removed - changes to reminder date, when hitting snooze
